chat-plugins/modlog.js

```diff
@@ -165,46 +165,21 @@ function runModlog(room, searchString, exactSearch, maxLines) {
 		fileNameList = fileNameList.map(filename => path.normalize(`${__dirname}/${LOG_PATH}${filename}`));
 		const searchStringRegex = new RegExp(regexString, 'i');
 		for (let i = 0; i < fileNameList.length; i++) {
-			checkRoomModlog(fileNameList[i], searchStringRegex, results);
+			await checkRoomModlog(fileNameList[i], searchStringRegex, results);
 		}
 	}
 	const resultData = results.getListClone();
 	return resultData.join('\n');
 }
 
-function checkRoomModlog(path, regex, results) {
-	if (!fs.existsSync(path)) return results;
-	let newLines = [];
-	let lineFragment = '';
-	let blockSize = 1024;
-	let buf = Buffer.alloc(blockSize);
-	const stats = fs.statSync(path);
-	let startPos = stats.size;
-	const fd = fs.openSync(path, 'r');
-
-	outerLoop: do {
-		startPos -= blockSize;
-		if (startPos < 0) {
-			blockSize += startPos;
-			startPos = 0;
-		}
-		const bytesRead = fs.readSync(fd, buf, 0, blockSize, startPos);
-		newLines = buf.toString().substr(0, bytesRead).split('\n').reverse();
-		newLines[0] += lineFragment;
-		lineFragment = newLines[newLines.length - 1];
-		if (startPos > 0) newLines.splice(-1, 1);
-		for (let i = 0; i < newLines.length; i++) {
-			if (newLines[i] && regex.test(newLines[i])) {
-				const insertionSuccessful = results.tryInsert(newLines[i]);
-				if (!insertionSuccessful) {
-					break outerLoop;
-				}
-			}
+async function checkRoomModlog(path, regex, results) {
+	const fileContents = await FS(path).read();
```

Comment:
`.read()` is for the most part unnecessary.

`.readTextIfExists()` does the same thing, but automatically does the string interpretation and doesn't crash if the file doesn't exist.